lib/pbench/agent/results.py

```diff
@@ -18,49 +17,61 @@ def __init__(self, result_dir, target_dir, user, prefix, config, logger):
         assert (
             config and logger
         ), f"config, '{config!r}', and/or logger, '{logger!r}', not provided"
-        if not result_dir:
-            logger.error("Result directory not provided")
-            sys.exit(1)
-        if not target_dir:
-            logger.error("Target directory not provided")
-            sys.exit(1)
-        try:
-            self.result_dir = Path(result_dir).resolve(strict=True)
-        except FileNotFoundError:
-            logger.error("Invalid result directory provided: {}", result_dir)
-            sys.exit(1)
-        else:
-            if not self.result_dir.is_dir():
-                logger.error("Invalid result directory provided: {}", result_dir)
-                sys.exit(1)
-        try:
-            self.target_dir = Path(target_dir).resolve(strict=True)
-        except FileNotFoundError:
-            logger.error("Invalid target directory provided: {}", target_dir)
-            sys.exit(1)
-        else:
-            if not self.target_dir.is_dir():
-                logger.error("Invalid target directory provided: {}", target_dir)
-                sys.exit(1)
+        self.result_dir = result_dir
+        self.target_dir = target_dir
         self.user = user
         self.prefix = prefix
         self.config = config
         self.logger = logger
 
+    def check_result_target_dir(self, result_dir, target_dir):
+        if not result_dir:
+            raise FileNotFoundError("Result directory not provided")
+        else:
+            try:
+                self.result_dir = Path(result_dir).resolve(strict=True)
+            except Exception:
+                raise FileNotFoundError(
+                    "Invalid result directory provided: {}", result_dir
+                )
+            else:
+                if not self.result_dir.is_dir():
+                    self.logger.error(
+                        "Invalid result directory provided: {}", result_dir
+                    )
+                    return 1
```

Comment:
Now we're raising exceptions on some errors and returning `1` on others, which is weird. Let's not do both!

Reply:
There's a [NotADirectoryError](https://docs.python.org/3/library/exceptions.html#NotADirectoryError) exception, if you like.

Reply:
Thanku :).. Done